<a href="https://colab.research.google.com/github/davivi-py/analise-acidentes-aereos-cenipa/blob/main/an%C3%A1lise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
!git clone https://github.com/davivi-py/analise-acidentes-aereos-cenipa

Cloning into 'analise-acidentes-aereos-cenipa'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 32 (delta 3), reused 9 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (32/32), 1.03 MiB | 14.83 MiB/s, done.
Resolving deltas: 100% (3/3), done.


In [4]:
import pandas as pd
import numpy as np
import sqlite3

ocorrencia = pd.read_csv('analise-acidentes-aereos-cenipa/data/ocorrencia.csv', encoding='latin-1', sep=';')
aeronave = pd.read_csv('analise-acidentes-aereos-cenipa/data/aeronave.csv', encoding='latin-1', sep=';')
ocorrencia_tipo = pd.read_csv('analise-acidentes-aereos-cenipa/data/ocorrencia_tipo.csv', encoding='latin-1', sep=';') # carregando as 3 tabelas

con = sqlite3.connect('db') # criação de banco de dados local usando sqlite3

ocorrencia.to_sql(name = "ocorrencia",con = con, if_exists =  'replace')   # colocando os 3 dataframes no banco de dados SQL
aeronave.to_sql(name = "aeronave",con = con, if_exists = 'replace')
ocorrencia_tipo.to_sql(name = "ocorrencia_tipo",con = con, if_exists ='replace')

10284

In [5]:
ocorrencia.shape  # inspecao inicial
ocorrencia.head()

,codigo_ocorrencia,codigo_ocorrencia1,codigo_ocorrencia2,codigo_ocorrencia3,codigo_ocorrencia4,ocorrencia_classificacao,ocorrencia_latitude,ocorrencia_longitude,ocorrencia_cidade,ocorrencia_uf,...,ocorrencia_dia,ocorrencia_hora,investigacao_aeronave_liberada,investigacao_status,divulgacao_relatorio_numero,divulgacao_relatorio_publicado,divulgacao_dia_publicacao,total_recomendacoes,total_aeronaves_envolvidas,ocorrencia_saida_pista
0,83097,83097,83097,83097,83097,ACIDENTE,-2.352.777.777.777,-4.563.944.444.444,PARAIBUNA,SP,...,31/12/2023,18:00:00,SIM,FINALIZADA,***,NÃO,NaN,0,1,NÃO
1,83085,83085,83085,83085,83085,INCIDENTE,-9.868.333.333.333,-6.789.805.555.555,RIO BRANCO,AC,...,13/12/2023,11:33:00,SIM,FINALIZADA,***,NÃO,NaN,0,1,NÃO
2,83074,83074,83074,83074,83074,INCIDENTE,-1.962.444.444.444,-4.397.194.444.444,BELO HORIZONTE,MG,...,20/12/2023,20:00:00,SIM,FINALIZADA,***,NÃO,NaN,0,1,NÃO
3,83063,83063,83063,83063,83063,INCIDENTE,-1.587.111.111.111,-4.791.861.111.111,BRASÍLIA,DF,...,29/12/2023,21:20:00,SIM,FINALIZADA,***,NÃO,NaN,0,1,NÃO
4,83051,83051,83051,83051,83051,INCIDENTE,-4.871.388.888.888,-668.975,CARAUARI,AM,...,09/12/2023,18:30:00,SIM,FINALIZADA,***,NÃO,NaN,0,1,NÃO


In [6]:
aeronave.shape
aeronave.head() # inspecao inicial

,codigo_ocorrencia2,aeronave_matricula,aeronave_operador_categoria,aeronave_tipo_veiculo,aeronave_fabricante,aeronave_modelo,aeronave_tipo_icao,aeronave_motor_tipo,aeronave_motor_quantidade,aeronave_pmd,...,aeronave_pais_fabricante,aeronave_pais_registro,aeronave_registro_categoria,aeronave_registro_segmento,aeronave_voo_origem,aeronave_voo_destino,aeronave_fase_operacao,aeronave_tipo_operacao,aeronave_nivel_dano,aeronave_fatalidades_total
0,83097,PRHDB,***,HELICÓPTERO,ROBINSON HELICOPTER,R44,R44,PISTÃO,MONOMOTOR,1088,...,BRASIL,BRASIL,HELICÓPTERO,PARTICULAR,CAMPO DE MARTE - SP,MAROUM,CRUZEIRO,***,SUBSTANCIAL,4
1,83085,PTEEK,***,AVIÃO,EMBRAER,EMB-810C,PA34,PISTÃO,BIMOTOR,2073,...,BRASIL,BRASIL,AVIÃO,TÁXI AÉREO,FAZENDA ÁGUA LIMPA,PLÁCIDO DE CASTRO,TÁXI,NÃO REGULAR,LEVE,0
2,83074,PSAEN,***,AVIÃO,EMBRAER,ERJ 190-400,E295,JATO,BIMOTOR,61200,...,BRASIL,BRASIL,AVIÃO,REGULAR,VIRACOPOS,TANCREDO NEVES,INDETERMINADA,REGULAR,NENHUM,0
3,83063,PRGZW,***,AVIÃO,BOEING COMPANY,737-86N,B738,NaN,BIMOTOR,79015,...,BRASIL,BRASIL,AVIÃO,REGULAR,GOVERNADOR ANDRÉ FRANCO MONTORO,PRESIDENTE JUSCELINO KUBITSCHEK,POUSO,REGULAR,NENHUM,0
4,83051,PRPDP,***,***,ATR - GIE AVIONS DE TRANSPORT RÉGIONAL,ATR-42-500,AT45,TURBOÉLICE,BIMOTOR,18600,...,BRASIL,BRASIL,***,REGULAR,PORTO URUCU,CARAUARI,TÁXI,REGULAR,NENHUM,0


In [7]:
ocorrencia_tipo.shape
ocorrencia_tipo.head() # inspecao inicial

,codigo_ocorrencia1,ocorrencia_tipo,ocorrencia_tipo_categoria,taxonomia_tipo_icao
0,83097,INDETERMINADO,INDETERMINADO,UNK
1,83085,FALHA OU MAU FUNCIONAMENTO DE SISTEMA / COMPON...,FALHA OU MAU FUNCIONAMENTO DE SISTEMA / COMPON...,SCF-NP
2,83074,COLISÃO COM AVE,COLISÃO COM AVE,BIRD
3,83063,COLISÃO COM AVE,COLISÃO COM AVE,BIRD
4,83051,COLISÃO COM AVE,COLISÃO COM AVE,BIRD


In [8]:
ocorrencia.isna().sum() # contagem de NaN (nulos)

,0
codigo_ocorrencia,0
codigo_ocorrencia1,0
codigo_ocorrencia2,0
codigo_ocorrencia3,0
codigo_ocorrencia4,0
ocorrencia_classificacao,0
ocorrencia_latitude,2581
ocorrencia_longitude,2581
ocorrencia_cidade,0
ocorrencia_uf,0


In [9]:
aeronave.isna().sum() # contagem de NaN (nulos)

,0
codigo_ocorrencia2,0
aeronave_matricula,0
aeronave_operador_categoria,0
aeronave_tipo_veiculo,0
aeronave_fabricante,1
aeronave_modelo,2
aeronave_tipo_icao,1
aeronave_motor_tipo,113
aeronave_motor_quantidade,0
aeronave_pmd,0


In [10]:
ocorrencia_tipo.isna().sum() # contagem de NaN (nulos)

,0
codigo_ocorrencia1,0
ocorrencia_tipo,0
ocorrencia_tipo_categoria,0
taxonomia_tipo_icao,0


# Investigando uma diferença de linhas após JOIN

Quando juntamos as tabelas *ocorrencia* e *aeronave*, o resultado tem **mais** linhas do que ocorrencia **sozinha**, o que nos leva a pensar, "nossa, será que tem alguma ocorrência sendo duplicada?"

In [11]:
query_sem_join = """SELECT COUNT(*) FROM ocorrencia"""
pd.read_sql_query(query_sem_join, con)

,COUNT(*)
0,9821


In [14]:
query_com_join = """SELECT COUNT(*) FROM ocorrencia JOIN aeronave ON aeronave.codigo_ocorrencia2 = ocorrencia.codigo_ocorrencia2"""
pd.read_sql_query(query_com_join, con)

,COUNT(*)
0,9926


Porém, ao analisar a coluna "total_aeronaves_envolvidas" da tabela ocorrencia, podemos perceber que existem casos em que mais de uma aeronave está associada a um único acontecimento.

In [17]:
query_aeronave_env = """ SELECT total_aeronaves_envolvidas FROM ocorrencia WHERE total_aeronaves_envolvidas > 1"""
pd.read_sql_query(query_aeronave_env, con)

,total_aeronaves_envolvidas
0,2
1,2
2,2
3,2
4,2
...,...
95,2
96,2
97,2
98,4


Os 100 casos com múltiplas aeronaves parecem explicar a diferença de linhas, mas será que explicam exatamente? Se a soma de total_aeronaves_envolvidas em todas as ocorrências bater com o total de linhas do JOIN (9926), confirma que não sobra nada sem explicação.

In [18]:
aeronaves_envolvidas = """ SELECT SUM(total_aeronaves_envolvidas) FROM ocorrencia"""
pd.read_sql_query(aeronaves_envolvidas, con)

,SUM(total_aeronaves_envolvidas)
0,9926


A soma bateu exatamente: 9926. Isso confirma que a diferença de linhas após o JOIN se explica inteiramente pelas ocorrências com múltiplas aeronaves, não há duplicação nem perda de dados no processo.

Mesmo sabendo que as linhas estão corretas, a coluna aeronave_fatalidades_total é per-aeronave ou total repetido?